avm/res/compute/gallery/main.bicep

```diff
@@ -220,4 +236,88 @@ type roleAssignmentType = {
 
   @sys.description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
-}[]?
+}[]
+
+type imageType = {
+  @sys.description('Required. Name of the image definition.')
+  @minLength(1)
+  @maxLength(80)
+  name: string
+
+  @sys.description('Optional. The description of this gallery image definition resource. This property is updatable.')
+  description: string?
+
+  @sys.description('Required. This property allows you to specify the type of the OS that is included in the disk when creating a VM from a managed image.')
+  osType: ('Linux' | 'Windows')
+
+  @sys.description('Required. This property allows the user to specify the state of the OS of the image.')
+  osState: ('Generalized' | 'Specialized')
+
+  @sys.description('Required. This is the gallery image definition identifier.')
+  identifier: {
```

Comment:
If you 'export()` the types in the child, you should be able to import them here just as you wanted to originally